functions/Set-DbaAgentJobStep.ps1

```diff
@@ -0,0 +1,332 @@
+                {
+                    # Check if the username is present in the database
+                    if(($Server.Databases[$DatabaseName].Users).Name -contains $DatabaseUserName)
+                    {
+                        Write-Message -Message "Setting job step database username to '$($DatabaseUserName)'" -Level 5 -Silent $Silent
+                        $JobStep.DatabaseUserName = $DatabaseUserName
+                    }
+                    else 
+                    {
+                        Stop-Function -Message ("The database user is not present in the database '$($DatabaseName)' on instance '$($SqlServer)'.") -Silent $Silent -Target $SqlServer 
+                        return
+                    }
+                }
+
+                if($RetryAttempts -ge 1)
+                {
+                    Write-Message -Message "Setting job step retry attempts to $($RetryAttempts)" -Level 5 -Silent $Silent
+                    $JobStep.RetryAttempts = $RetryAttempts
+                }
+
+                if($RetryInterval -ge 1)
+                {
+                    Write-Message -Message "Setting job step retry interval to $($RetryInterval)" -Level 5 -Silent $Silent
+                    $JobStep.RetryInterval = $RetryInterval
+                }
+
+                if($OutputFileName.Length -ge 1)
+                {
+                    Write-Message -Message "Setting job step output file name to '$($OutputFileName)'" -Level 5 -Silent $Silent
+                    $JobStep.OutputFileName = $OutputFileName
+                }
+
+                if($ProxyName.Length -ge 1)
+                {
+                    # Check if the proxy exists
+                    if(($Server.JobServer.ProxyAccounts).Name -contains $ProxyName)
+                    {
+                        Write-Message -Message "Setting job step proxy name to $($ProxyName)" -Level 5 -Silent $Silent
+                        $JobStep.ProxyName = $ProxyName
+                    }
+                    else 
+                    {
+                        Stop-Function -Message ("The proxy name '$($ProxyName)' doesn't exist on instance '$($SqlServer)'.") -Silent $Silent -Target $SqlServer 
+                        return
+                    }
+                }
+                #region job step options
+
+                # Execute 
+                if($PSCmdlet.ShouldProcess($SqlServer, ("Changing the job step $($StepName)"))) 
+                {
+                    try
+                    {
+                        Write-Message -Message ("Changing the job step") -Level 2 -Silent $Silent
+                        
+                        # Change the job step
+                        $JobStep.Alter()
+                    }
+                    catch
+                    {
+                        Write-Message -Message ("Something went wrong changing the job step. `n$($_.Exception.Message)") -Level 2 -Silent $Silent 
```

Comment:
Instead of writing a warning, use stop-function here. Even if it's the last line and nothing comes after it, Stop-Function exists to handle unattended mode, for example as part of scripts. If this fails, the calling script must now, so it needs the exception thrown by Stop-Function in silent mode. Even if not, it handles debugging information and should be used for troubleshooting and support information.